### ISSUE TITLE
Unable to retrieve Management API access token due to change in 8.4.0

### ISSUE THREAD

**Issue by chrisnetonline** [auth0/auth0-PHP#705]
### Checklist

- [X] The issue can be reproduced in a a sample app (either [webapp](https://github.com/auth0-samples/auth0-php-web-app) or [api](https://github.com/auth0-samples/auth0-php-api-samples)) (or N/A).
- [X] I have looked into the [README](https://github.com/auth0/auth0-PHP#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/auth0-PHP/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-PHP/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-PHP/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/tag/auth0-php) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### SDK Version

8.3

### PHP Version

PHP 8.1

### Description

This is a regression in the new 8.4 SDK. 

The new `addClientAuthentication` method introduced in [feat(Authentication): Add support for Client Assertion](https://github.com/auth0/auth0-PHP/pull/699/files#top) #699 is overwriting the Management API `client_id` and `client_secret` with the Authentication `client_id` and `client_secret` set in the following sample code:

```
    protected function obtainManagementToken(Auth0Interface $client): void
    {
        $response = $client->authentication()->clientCredentials([
            'audience' => 'https://' . $this->config['domain'] . '/api/v2/',
            'client_id' => $this->config['management']['client_id'],
            'client_secret' => $this->config['management']['client_secret'],
        ]);

        if ($response->getStatusCode() >= 300) {
            Log::error('Auth0AuthorizationService@obtainManagementToken: Unexpected response', [
                'responseBody' => $response->getBody()->getContents(),
                'responseCode' => $response->getStatusCode(),
            ]);

            throw new RequestException(
                message: 'Unexpected status code: ' . $response->getReasonPhrase(),
                code: $response->getStatusCode(),
            );
        }

        $responseData = json_decode($response->getBody());

        $client->configuration()->setManagementToken($responseData?->access_token);
    }

    protected function getClient(): Auth0Interface
    {
        return new Auth0([
            'domain' => $this->config['domain'],
            'clientId' => $this->config['client_id'],
            'clientSecret' => $this->config['client_secret'],
            'cookieSecret' => $this->config['cookie_secret'],
            'cookieDomain' => $this->config['cookie_domain'],
        ]);
    }
```

This works fine in 8.3.8 and is broken in 8.4.0.

### How can we reproduce this issue?

See notes above.

**Comment by evansims** [comment]
Hey @chrisnetonline 👋 Thanks very much for bringing this to my attention! I'll get a fix prepared.

**Comment by evansims** [comment]
Hi @chrisnetonline 👋 Thanks for being patient, the team has been at an offsite this past week. We've prepared a fix that will ship with 8.5.0 this week.
